woocommerce/woocommerce
# Fix husky git hooks

### All Submissions:






### Changes proposed in this Pull Request:



Closes #32399.

Fixes pre-commit, pre-push, post-merge hooks

- As recommended [here](https://typicode.github.io/husky/#/?id=recipes), add husky and lint-staged in root directory.
- Removed husky and lint-staged in `plugins/*`
- Add composer.json in the root directory to run phpcs command
- Add `pnpm git:update-hooks` to `postinstall` to automatically have Git hooks enabled after install



### How to test the changes in this Pull Request:

1.  Checkout a test branch  `git checkout -b test/32399`
2.  Modify a JS file and a PHP file
3. Commit the changes
4. Should trigger pre-commit hook
5. Change `PROTECTED_BRANCH` in `bin/pre-push.sh` to `test/32399`
6. Run `git push --set-upstream origin test/32399`
7. Should trigger pre-push hook
8. Create another branch  `git checkout -b test/32399-new`
9. Run `pnpm add -w foo` to change `pnpm-lock.yaml` & commit this change
10. Checkout back to `test/32399`
11. Merge `test/32399-new`
12. Should trigger post-merge hook (automatically run `pnpm install`)

### Other information:




### Changelog entry

Dev: Fix husky git hooks

### FOR PR REVIEWER ONLY:
